cfgov/v1/migrations/0136_migrate_imagetextgroups_to_infounitgroup.py

```diff
@@ -0,0 +1,278 @@
+                    'level': 'h3',
+                }
+
+            # If at least one image_text has checked has_rule,
+            # assume we want them between all
+            if image_text.get('has_rule'):
+                new_value['lines_between_items'] = True
+
+            # Fields that need no manipulation: body, image, links.
+            if image_text.get('body'):
+                info_unit['body'] = image_text['body']
+            if image_text.get('image'):
+                info_unit['image'] = image_text['image']
+            if image_text.get('links'):
+                info_unit['links'] = image_text['links']
+            else:
+                info_unit['links'] = []
+
+            # Fields being discarded: is_widescreen, is_button.
+
+            # If at least one field was copied, append new info_unit to list.
+            if info_unit:
+                info_units.append(info_unit)
+
+        new_value['info_units'] = info_units
+
+    # The one field that needs no manipulation: sharing.
+    if old_value.get('sharing'):
+        new_value['sharing'] = old_value['sharing']
+
+    # This one has to be more explicit because it now defaults to True,
+    # but we don't want it to be True if it wasn't set before.
+    if old_value.get('link_image_and_heading'):
+        new_value['link_image_and_heading'] = True
+    else:
+        new_value['link_image_and_heading'] = False
+
+    # Add new required field: format.
+    new_value['format'] = new_format
+
+    return new_value
+
+
+def migrate_streamfield_forward(page_or_revision, streamfield_name, new_type):
+    """ Migrate a StreamField belonging to the page or revision """
+    old_stream_data = get_stream_data(page_or_revision, streamfield_name)
+
+    new_stream_data = []
+    migrated = False
+
+    block_conversions = {
+        'image_text_25_75_group': '25-75',
+        'image_text_50_50_group': '50-50',
+    }
+
+    for block in old_stream_data:
+        block_type = block['type']
+
+        if block_type in block_conversions:
+            new_block = {
+                'type': new_type,
+                'value': image_text_group_to_info_unit_group(
+                    block['value'],
+                    block_conversions[block_type]
+                )
+            }
+
+            new_stream_data.append(new_block)
+            migrated = True
+        else:
+            new_stream_data.append(block)
+
+    if migrated:
+        set_stream_data(
+            page_or_revision,
+            streamfield_name,
+            new_stream_data
+        )
+
+
+def forwards(apps, schema_editor):
+    models_and_fields = [
+        ('v1', 'BlogPage', 'content'),
+        ('v1', 'BrowsePage', 'content'),
+        ('v1', 'LandingPage', 'content'),
+        ('v1', 'LearnPage', 'content'),
+        ('v1', 'SublandingPage', 'content'),
+    ]
+
+    revision_model = apps.get_model('wagtailcore.PageRevision')
+
+    for app, page_type, streamfield_name in models_and_fields:
```

Comment:
There's quite a bit of code here that's duplicated between these two migrations and https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/util/migrations.py. I realize why — the need to pass `new_type` based on the page mode... but I can't help feeling that `migrate_page_types_and_fields()` could be used with `partial` passing `new_type`. Something akin to:

```python
from functools import partial

def migrate_streamfield_forward(page_or_revision, data, new_type=None):
    …

def forwards(apps, schema_editor):
    other_page_mapper = partial(migrate_streamfield_forward, new_type=info_unit_group)
    learn_page_mapper = partial(migrate_streamfield_forward, new_type=info_unit_group_25_75_only)
    migrate_page_types_and_fields(
        apps,
        [
            ('v1', 'DocumentDetailPage', 'sidefoot', 'related_metadata'),
            ('v1', 'BlogPage', 'content'),
            ('v1', 'BrowsePage', 'content'),
            ('v1', 'LandingPage', 'content'),
            ('v1', 'SublandingPage', 'content'),
        ],
        other_page_mapper
    )
    migrate_page_types_and_fields(
        apps,
        [
            ('v1', 'LearnPage', 'content'),
        ],
        learn_page_mapper
    )
```

But I don't feel strongly enough to recommend you re-work all of this. It's good as it is.